iroha/src/smartcontracts/isi/mod.rs

```diff
@@ -60,9 +60,9 @@ pub enum FindError {
     #[cfg(feature = "roles")]
     #[error("Failed to find role by id")]
     Role(RoleId),
-    /// Block with supplied hash not found.
-    #[error("Failed to find block with this hash")]
-    Block(Hash),
+    /// Block with supplied parent hash not found. More description in a string.
+    #[error("Block not found")]
+    Block(String),
```

Comment:
I would advice to create HashNotFound struct e.g. (`struct HashNotFound(Hash)`) for which to implement the display with this message.

Why do this:
1. In the case we need to match and destructure and handle errors in iroha.
2. The point above also applies to clients.
3. We will need to add translations later to various languages.

Reply:
We initially had errors as String, then we gradually started switching to structured errors as the project became more mature.

Reply:
Like this?

Reply:
> 
> 
> We initially had errors as String, then we gradually started switching to structured errors as the project became more mature.

But, as it seems now, structured errors don't give you the flexibility of string-formatting.

Reply:
Yes, like this

Reply:
They do as we can implement Display for them in any way we plan for the use cases that need string and use cases that need struct and raw hash - will use it.

Reply:
If we implement Display for all such error types then this code will be not needed:
https://github.com/hyperledger/iroha/blob/iroha2-dev/iroha_error/iroha_error_macro/src/display.rs#L29

Reply:
It is still needed because it glues together nested displays

Reply:
> 
> 
> It is still needed because it glues together nested displays

Yes, now. But I understood that you proposed to implement Display for all variants of that enum.

Reply:
Yes and then the macro will glue together displays for inner structs and variant prefixes.

Reply:
In general yes, it will be better to have `thiserror` like formatting

Reply:
Then, we finish this task as it is, and then in another task we move all this lewdness to `thiserror`?

Reply:
Yes, let's proceed this way